sdk/storage/src/common/crypt.cpp

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "common/crypt.hpp"
+
+#include "common/storage_common.hpp"
+
+#ifdef _WIN32
+#include <Windows.h>
+#include <bcrypt.h>
+#else
+#include <openssl/bio.h>
+#include <openssl/buffer.h>
+#include <openssl/evp.h>
+#include <openssl/hmac.h>
+#endif
+
+#include <stdexcept>
+
+namespace Azure { namespace Storage {
+
+#ifdef _WIN32
+  std::string HMAC_SHA256(const std::string& text, const std::string& key)
+  {
+    struct AlgorithmProviderInstance
+    {
+      BCRYPT_ALG_HANDLE Handle;
+      std::size_t ContextSize;
+      std::size_t HashLength;
+
+      AlgorithmProviderInstance()
+      {
+        NTSTATUS status = BCryptOpenAlgorithmProvider(
+            &Handle, BCRYPT_SHA256_ALGORITHM, NULL, BCRYPT_ALG_HANDLE_HMAC_FLAG);
+        if (!BCRYPT_SUCCESS(status))
+        {
+          throw std::runtime_error("failed to call bcrypt function");
+        }
+        DWORD objectLength = 0;
+        DWORD dataLength = 0;
+        status = BCryptGetProperty(
+            Handle,
+            BCRYPT_OBJECT_LENGTH,
+            (PBYTE)&objectLength,
+            sizeof(objectLength),
+            &dataLength,
+            0);
+        if (!BCRYPT_SUCCESS(status))
+        {
+          throw std::runtime_error("failed to call bcrypt function");
+        }
+        ContextSize = objectLength;
+        DWORD hashLength = 0;
+        status = BCryptGetProperty(
+            Handle, BCRYPT_HASH_LENGTH, (PBYTE)&hashLength, sizeof(hashLength), &dataLength, 0);
+        if (!BCRYPT_SUCCESS(status))
+        {
+          throw std::runtime_error("failed to call bcrypt function");
+        }
+        HashLength = hashLength;
+      }
+
+      ~AlgorithmProviderInstance() { BCryptCloseAlgorithmProvider(Handle, 0); }
+    };
+
+    static AlgorithmProviderInstance AlgorithmProvider;
+
+    std::string context;
+    context.resize(AlgorithmProvider.ContextSize);
```

Comment:
This can be a one liner right?

Reply:
all the code is automatically formatted by the .clang-format file, which is shared among all the sdks in this repo. I think there's nothing wrong to follow that one. If you don't like one specific style, please bring it up in weekly meeting, so that we can change the clang-format file.